crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -265,11 +289,16 @@ impl UiSurface {
         }
     }
 
-    /// Removes each entity from the internal map and then removes their associated node from taffy
+    /// Removes each entity from the internal map and then removes their associated nodes from taffy
     pub fn remove_entities(&mut self, entities: impl IntoIterator<Item = Entity>) {
         for entity in entities {
             if let Some(node) = self.entity_to_taffy.remove(&entity) {
-                self.taffy.remove(node).unwrap();
+                self.taffy.remove(node.id).unwrap();
+                if let Some(viewport_node) = node.viewport_id {
+                    if self.taffy.get_node_context(viewport_node).is_some() {
+                        self.taffy.remove(viewport_node).ok();
+                    }
```

Comment:
This feels silly but I couldn't find a simpler way to check for existence before removal.

Reply:
Maybe `let _ = self.taffy.remove(viewport_node);`?

Reply:
The problem is  thateven though `remove` returns a result it doesn't return `Err` on failure, it just panics